src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -63,84 +68,93 @@ public class ExprTarget extends PropertyExpression<LivingEntity, Entity> {
-		return type != null ? type.getType() : Entity.class;
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(LivingEntity.class);
+		}
+		return super.acceptChange(mode);
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		if (e == null)
-			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
-		return Classes.getDebugMessage(getAll(e));
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.RESET || mode == ChangeMode.DELETE) { // null will make the entity target-less (reset target) but for players it will remove them
+			LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
+			for (LivingEntity entity : getExpr().getArray(event)) {
+				if (event instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+					((EntityTargetEvent) event).setTarget(target);
+				} else {
+					if (entity instanceof Mob) {
+						((Mob) entity).setTarget(target);
+					} else if (entity instanceof Player && mode == ChangeMode.DELETE) {
+						Entity playerTarget = getTarget(entity, type);
+						if (playerTarget != null && !(playerTarget instanceof OfflinePlayer))
+							playerTarget.remove();
+					}
+				}
+			}
+		}
+		super.change(event, delta, mode);
 	}
 
 	@Override
 	public boolean setTime(int time) {
-		return super.setTime(time, EntityTargetEvent.class, getExpr());
+		return time >= 0;
 	}
 
 	@Override
-	@Nullable
-	public Class<?>[] acceptChange(ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
-			return CollectionUtils.array(LivingEntity.class);
-		return super.acceptChange(mode);
+	public Class<? extends Entity> getReturnType() {
+		return type != null ? type.getType() : Entity.class;
 	}
 
 	@Override
-	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
-			LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
-			for (LivingEntity entity : getExpr().getArray(e)) {
-				if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					((EntityTargetEvent) e).setTarget(target);
-				} else if (entity instanceof Mob) {
-					((Mob) entity).setTarget(target);
-				}
-			}
-			return;
-		}
-		super.change(e, delta, mode);
+	public String toString(@Nullable Event event, boolean debug) {
+		if (event == null)
+			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(event, debug));
+		return Classes.getDebugMessage(getAll(event)); // TODO for players it will return null (need to getTarget)
```

Comment:
This is a debug message method. It shouldn't be returning a list of the used entities from the syntax. That provides no information as to what syntax it is from. It should return a normal toString like you're used to.